src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -47,6 +48,77 @@ protected static class ArtifactEventProperties
         }
     }
 
+    // Caller: PublishPipelineArtifact task
+    // Can be invoked from a build run or a release run should a build be set as the artifact. 
+    public class PublishPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
+    {
+        public override Guid Id => PipelineArtifactPluginConstants.PublishPipelineArtifactTaskId;
+        protected override string TargetPath => "path";
+
+        public static readonly Regex jobIdentifierRgx = new Regex("[^a-zA-Z0-9 - .]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        protected override async Task ProcessCommandInternalAsync(
+            AgentTaskPluginExecutionContext context, 
+            CancellationToken token)
+        {           
+            string artifactName = context.GetInput(ArtifactEventProperties.ArtifactName, required: false);
+            string targetPath = context.GetInput(TargetPath, required: true);
+            string defaultWorkingDirectory = context.Variables.GetValueOrDefault("system.defaultworkingdirectory").Value;
+
+            targetPath = Path.IsPathFullyQualified(targetPath) ? targetPath : Path.GetFullPath(Path.Combine(defaultWorkingDirectory, targetPath));
+
+            string hostType = context.Variables.GetValueOrDefault("system.hosttype")?.Value; 
+            if (!string.Equals(hostType, "Build", StringComparison.OrdinalIgnoreCase)) {
+                throw new InvalidOperationException(
+                    StringUtil.Loc("CannotUploadFromCurrentEnvironment", hostType ?? string.Empty)); 
+            }
+
+            if (String.IsNullOrWhiteSpace(artifactName))
+            {
+                string jobIdentifier = context.Variables.GetValueOrDefault("system.jobIdentifier").Value;
+                var normalizedJobIdentifier = NormalizeJobIdentifier(jobIdentifier);
+                artifactName = normalizedJobIdentifier;
+            }
+
+            if(!PipelineArtifactPathHelper.isValidPath(artifactName)) {
+                throw new ArgumentException(StringUtil.Loc("ArtifactNameIsNotValid"));
```

Comment:
May be a better error message would be to tell the user which all chars are invalid? Something like - "Artifact name should not containt /,\,|..."  🤔

Reply:
I thought about it but then noticed that all the other messages are this kind of style since it's actually a label for localization.

Reply:
I think @willsmythe may have more knowledge on who willing be updating these strings into different languages?

Reply:
I added the en-US string for this error message.